Modify wc_get_low_stock_amount function to always return a number

Konamiman opened the pull request:
### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The `wc_get_low_stock_amount` function is documented as returning an `int`, but when a product didn't have an explicit low stock amount configured the default value from the `woocommerce_notify_low_stock_amount` was returned verbatim, and that value is stored as a string. This pull request makes sure that the function always returns a number.

Also the related unit tests are updated so that they set the value of the `woocommerce_notify_low_stock_amount` option as a string, which is what actually happens in WooCommerce; and now they also verify that the function returns a number.

Closes #29525.

### How to test the changes in this Pull Request:

The easiest way to test this is by using the wp cli tool.

1. Create/edit a product so that it has "Enable stock management at product level" set and "Low stock threshold" unset:

![image](https://user-images.githubusercontent.com/937723/115363338-a93ce600-a1c2-11eb-880a-20932ddf900e.png)

2. Take note of the product id, it's in the product edit page url (`?post=...`) or in the products list (`ID: ...` when you hover the product row).

3. Run this in a console: 

```php
wp eval 'echo gettype(wc_get_low_stock_amount(wc_get_product(<product id>)));'
```

Without the fix you'll get `string`, with the fix you'll get `integer`.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - wc_get_low_stock_amount was returning a string, not an integer, for products not having an explicit low stock value set.

Konamiman committed:
Fix code sniffer violations in stock functions and its tests

Konamiman committed:
Modify wc_get_low_stock_amount so that it always returns an integer.

Previously, if the product didn't have an explicit low stock value
amount the value of the woocommerce_notify_low_stock_amount option,
which is a string, was returned verbatim.

Also, update related unit tests to create the option value as a string,
and to check that the value returned by woocommerce_notify_low_stock_amount
is always an integer.

roykho commented on includes/wc-stock-functions.php:
```diff
@@ -403,13 +403,13 @@ function wc_get_low_stock_amount( WC_Product $product ) {
 	}
 
-	return $low_stock_amount;
+	return intval( $low_stock_amount );
```
Unless we're using the second argument to change the `base`, it is preferred to use `(int)` instead as it is faster than calling a type casting function.

roykho reviewed (changes requested):
Left a comment and also the unit test is failing.

Konamiman committed:
Use int cast instead of intval in wc_get_low_stock_amount

Konamiman committed:
Rename "assertIsInt" to "assertIsInteger" and make it static

- Renaming to prevent conflicts with the existing method in
  the newer PHPUnit used in PHP 8.
- Making it static because "assertIsInt" is static too, so it'll be
  easier to replace in the future.

roykho reviewed (approved):
LGTM! e2e tests are failing but upon looking at it, it doesn't seem like it is caused by this PR.
